### PR TITLE
kotlin 1.8.10 and friends

### PR DESCRIPTION
**Pull request by RBusarow**
- kotlin `1.8.10`
- ksp `1.8.10-1.0.9`
- drop Kotlinter because it doesn't play nicely with KSP's new source sets
- replace Kotlinter with `com.rickbusarow.ktlint`
- KtLint `0.49.1`
  - this comes with updated `.editorconfig` requirements

### CODE REVIEW

File: samples/containers/hello-back-button/src/main/java/com/squareup/sample/hellobackbutton/HelloBackButtonWorkflow.kt [hunk 1]
```diff
@@ -17,7 +17,7 @@ object HelloBackButtonWorkflow : StatefulWorkflow<Unit, State, Nothing, HelloBac
   enum class State : Parcelable {
     Able,
     Baker,
-    Charlie;
+    Charlie
```
Comment (steve-the-edwards) [inline samples/containers/hello-back-button/src/main/java/com/squareup/sample/hellobackbutton/HelloBackButtonWorkflow.kt:20]: whoa - no more `;` at the end. Can we have trailing commas :)?

File: workflow-ui/core-android/src/androidTest/java/com/squareup/workflow1/ui/container/fixtures/BackStackContainerLifecycleActivity.kt [hunk 2]
```diff
@@ -40,7 +40,7 @@ internal class BackStackContainerLifecycleActivity : AbstractLifecycleTestActivi
       context: Context,
       container: ViewGroup?
     ): ScreenViewHolder<BaseRendering> =
-      ScreenViewHolder(initialEnvironment, View(context)) { _, _ -> /* Noop */ }
+      ScreenViewHolder(initialEnvironment, View(context)) { _, _ -> }
```
Comment (steve-the-edwards) [inline workflow-ui/core-android/src/androidTest/java/com/squareup/workflow1/ui/container/fixtures/BackStackContainerLifecycleActivity.kt:43]: I don't get these? Did ktlint remove these now?

**Review by steve-the-edwards — APPROVED** [review]
🍾

File: workflow-ui/core-android/src/androidTest/java/com/squareup/workflow1/ui/container/fixtures/BackStackContainerLifecycleActivity.kt (hunk 2 above)
Comment (RBusarow) [reply workflow-ui/core-android/src/androidTest/java/com/squareup/workflow1/ui/container/fixtures/BackStackContainerLifecycleActivity.kt:43]: https://github.com/pinterest/ktlint/issues/1980

For the KtLint `0.49.x` season, the `comment-wrapping` rule doesn't allow for single-line block comments.  This is extracted into its own rule for `0.50.0`.

File: samples/containers/hello-back-button/src/main/java/com/squareup/sample/hellobackbutton/HelloBackButtonWorkflow.kt (hunk 1 above)
Comment (RBusarow) [reply samples/containers/hello-back-button/src/main/java/com/squareup/sample/hellobackbutton/HelloBackButtonWorkflow.kt:20]: This is the `standard:no-semi` rule.  I can enable it if you'd like me to. ;)

For commas... no, it's not possible.  It just doesn't work.  Lots of performance regressions and all of the internal canary branches crashed.

**Pull request merged**